database/rrdcalc.h

```diff
@@ -139,14 +139,23 @@ struct rrdcalc {
     struct rrdcalc *next;
 };
 
+struct rrdcalc_rrdset_alarm {
+    avl avl1;
+
+    RRDSET *st;
+    avl_tree_lock rrdvar_alarm_name_index;
+
+    struct rrdcalc_rrdset_alarm *next;
```

Comment:
I don't see `next` being used anywhere in the code, other than iterating for `alarm_index_link()` and `alarm_index_unlink_and_free()` but I don't see how they're doing something with it.

Reply:
Clion is not showing usage, but you can find the usage inside rrdhost in the following functions:

- `alarm_index_link`
- `alarm_index_unlink_and_free`

Reply:
But they're not doing something with those functions.

Reply:
`avl_remove_lock` does all the work anyway.

Reply:
`avl_remove_lock` removes the link inside the tree, the functions were created to free the structure that was allocated before.

Reply:
Thank you for the explanation, it was completely removed with the last commit.